### DISCUSSION TITLE
Emacs 29.1 fresh install - can't detect Typescript

### DISCUSSION THREAD

**Discussion by MarcCoquand** [joaotavora/eglot#1259] [Q&A] [answered]
Hey! 

I'm trying to understand how eglot is supposed to work. I've just installed emacs 29.1, my package list is only evil mode. From the manual it seems that it should be as simple as `M-x eglot` to start typescript however I'm running into the issue `[eglot] Couldn't guess LSP server for 'fundamental-mode'`. I have typescript language server installed and it's working for all other editors I'm using 😓 . Any clue what I am missing?

**Comment by manuel-uberti** [comment]
In order to understand the actual problem, could you try following the instructions at https://joaotavora.github.io/eglot/#Troubleshooting-Eglot?

**Comment by MarcCoquand** [comment, marked as answer]
Hey!

So I found out what it was due to, I had not enabled tree-sitter
properly. I needed to add 

```
(add-to-list 'auto-mode-alist '("\\.ts\\'" . typescript-ts-mode)
```

for tree-sitter to set the major mode. After that I could run `eglot`
and it would work.
